src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -27,6 +34,42 @@ internal class RPackageManager : IRPackageManager {
             return VisualComponent;
         }
 
+        public async Task<IList<RPackage>> GetInstalledPackagesAsync() {
+            if (_interactiveWorkflow.RSession == null || !_interactiveWorkflow.RSession.IsHostRunning) {
+                return new List<RPackage>();
+            }
+
+            REvaluationResult result;
+            using (IRSessionEvaluation eval = await _interactiveWorkflow.RSession.BeginEvaluationAsync()) {
```

Comment:
This might be one of those cases where it would be better to call `EvaluateAsync` directly on `IRSession` instead of going through `BeginEvaluationAsync`, since you don't care when it happens, or what the context is. Avoiding `BeginEvaluationAsync` means that it'll work even when REPL is running code.